Makes deadchat control inputs bypass automute.

TemporalOroboros opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Makes the input phrases for the deadchat control component bypass the automute for repeated identical messages.
It's slightly annoying when you get muted because you spent a minute trying to get the supermatter to move down.
It's also probably annoying to keep getting tickets requesting unmutes because of a TC trade you granted.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Gets rid of a minor annoyance for both players and admins.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
qol: The deadchat controllable component is no longer automute bait.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

TemporalOroboros committed:
Makes deadchat control inputs bypass automute.

It's slightly annoying when you get muted because you spent a minute trying to get the supermatter to move down.

TemporalOroboros commented:
Whoops.

Colovorat commented:
Holy shit finally, soul

Mothblocks commented on code/__DEFINES/dcs/signals/signals_client.dm:
```diff
@@ -0,0 +1,4 @@
+/// from base of [/client/proc/handle_spam_prevention] (message, mute_type)
+#define COMSIG_CLIENT_AUTOMUTE_CHECK "client_automute_check"
+/// Prevents the automute system checking this client for repeated messages.
+#define SIGVAL_WAIVE_AUTOMUTE_CHECK (1<<0)
```
1. Indent this up one so it's obvious it's part of `COMSIG_CLIENT_AUTOMUTE_CHECK`.
2. Typo, "SIGVAL"

TemporalOroboros replied:
2 is intentional. We have COMSIG -> COMponent SIGnal. I am adding SIGVAL -> SIGnal VALue.

Mothblocks replied:
COMSIG needs to be killed into SIGNAL anyway, that's from a time when signals could only be registered from components.

Can you just remove `SIGVAL_` entirely and just have `WAIVE_AUTOMUTE_CHECK`?

TemporalOroboros replied:
Sure.

Mothblocks commented on code/modules/client/client_procs.dm:
```diff
@@ -166,6 +166,9 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 
 
 	if(CONFIG_GET(flag/automute_on) && !holder && last_message == message)
+		if(SEND_SIGNAL(src, COMSIG_CLIENT_AUTOMUTE_CHECK, last_message, mute_type) & SIGVAL_WAIVE_AUTOMUTE_CHECK)
```
It might honestly be better to just do this on the mob so you don't have to do all that login stuff, but it's fine if you don't want to.

TemporalOroboros replied:
That's probably easier. I think I'll try that.

TemporalOroboros committed:
Does Mothblocks' suggestions.

Routes the automute check signal through the mob.
- This lets us get rid of the extra login signals.

Renames the automute check signal and its associated return flag.

Mothblocks commented on code/__DEFINES/dcs/signals/signals_client.dm:
```diff
@@ -0,0 +1,4 @@
+/// from base of [/client/proc/handle_spam_prevention] (message, mute_type)
+#define SIGNAL_CLIENT_AUTOMUTE_CHECK "client_automute_check"
```
I wasn't clear, sorry--keep this to COMSIG, renaming them all to SIGNAL is something I want to do at some point but should probably be done all at once.

TemporalOroboros replied:
Alright, done.

TemporalOroboros committed:
Regress to COMSIG

Converts SIGNAL_CLIENT_AUTOMUTE_CHECK to COMSIG_CLIENT_AUTOMUTE_CHECK for consistency with the other signals.

Mothblocks commented on code/__DEFINES/dcs/signals/signals_client.dm:
```diff
@@ -0,0 +1,4 @@
+/// from base of [/client/proc/handle_spam_prevention] (message, mute_type)
+#define COMSIG_CLIENT_AUTOMUTE_CHECK "client_automute_check"
```
Rename to `MOB_`, and put it in the mob signals file.

TemporalOroboros replied:
Done.

TemporalOroboros committed:
Moves the automute signal to the mob signal file.
